bugsnag-android-core/src/main/java/com/bugsnag/android/Exceptions.java

```diff
@@ -20,23 +23,8 @@
 
     @Override
     public void toStream(@NonNull JsonStream writer) throws IOException {
-        writer.beginArray();
-
-        // Unwrap any "cause" exceptions
-        Throwable currentEx = exception;
-        while (currentEx != null) {
-            if (currentEx instanceof JsonStream.Streamable) {
-                ((JsonStream.Streamable) currentEx).toStream(writer);
-            } else {
-                String exceptionName = getExceptionName(currentEx);
-                String localizedMessage = currentEx.getLocalizedMessage();
-                StackTraceElement[] stackTrace = currentEx.getStackTrace();
-                exceptionToStream(writer, exceptionName, localizedMessage, stackTrace);
-            }
-            currentEx = currentEx.getCause();
-        }
-
-        writer.endArray();
+        exception.setProjectPackages(projectPackages);
+        exception.toStream(writer);
```

Comment:
I need to spend a bit more time understanding the interactions between `Exceptions` and `BugsnagException` but so far its a little confusing how `BugsnagException` now represents an array rather than just the individual exception value, and in particular might introduce a situation where the "cause" exception of a `Streamable` might be skipped from serialization.

The deleted logic in `Exceptions` may have been more correct and simplify the execution path between `BugsnagException.toStream`, `BugsnagException.serializeException`, and `BugsnagException.writeThrowable`.

Reply:
Looking back at the implementation this makes sense. I've updated the logic so that `Exceptions` is responsible for iterating through the throwable causes, and made `BugsnagException` responsible for serializing its own information only.